client/tests/configuration_test.py

```diff
@@ -54,6 +56,7 @@ def test_init(
             },
             {},
         ]
+        expanduser.side_effect = ["a"]
```

Comment:
One neat trick with mock is that you can actually make side effects of mocked functions into lambdas. I have a suspicion this might save us a bit of code in this case :) Looks like we're consistently expanding "~" into "/home/user" in the below tests, so this code suggestion might save you the effort of writing all those side effects below.
```suggestion
        expanduser.side_effect = lambda path: path.replace("~", "/home/user")
```

Reply:
Very solid point, wish I'd recognized that sooner! Thank you! That's sufficient for all of the configuration testing 🎉